src/scripts/main.js

```diff
@@ -1,3 +1,322 @@
 'use strict';
 
-// write code here
+const table = document.querySelector('table');
+const tBody = document.querySelector('tbody');
+
+table.addEventListener('click', (e) => {
+  const employees = tBody.querySelectorAll('tr');
+
+  if (e.target.tagName === 'TD') {
+    const hasActive = [...employees]
+      .find(element => element.classList.contains('active'));
+
+    if (hasActive) {
+      hasActive.removeAttribute('class');
+    }
+
+    const tr = e.target.closest('tr');
+
+    tr.classList.add('active');
+  }
+
+  if (e.target.tagName === 'TH') {
+    sortGrid(e.target.cellIndex, [...employees], e.target);
+  }
+});
+
+document.addEventListener('click', (e) => {
+  const employees = tBody.querySelectorAll('tr');
+
+  const hasActive = [...employees]
+    .find(element => element.classList.contains('active'));
+
+  if (e.target.tagName === 'BODY' && hasActive) {
+    hasActive.removeAttribute('class');
+  }
+});
+
+function sortGrid(index, array, item) {
+  if (item.direction === 'ASC') {
+    item.direction = 'DESC';
+
+    array.sort((a, b) => {
+      const first = a.cells[index].innerText.replace(/[$,]/g, '');
+      const second = b.cells[index].innerText.replace(/[$,]/g, '');
+
+      if (isNaN(first)) {
+        return first.localeCompare(second);
+      }
+
+      return first - second;
+    });
+  } else {
+    item.direction = 'ASC';
+
+    array.sort((a, b) => {
+      const first = a.cells[index].innerText.replace(/[$,]/g, '');
+      const second = b.cells[index].innerText.replace(/[$,]/g, '');
+
+      if (isNaN(first)) {
+        return second.localeCompare(first);
+      }
+
+      return second - first;
+    });
```

Comment:
It looks like very similar part of code. Can we create separate function to do this?